File: bms/go/go-edlib/QgramDistance/QgramDistance.go
package test 

import (
    "math"
)

var (
)

const (
)

func QgramDistance(str1, str2 string, splitLength int) int {
	splittedStr1 := Shingle(str1, splitLength)
	splittedStr2 := Shingle(str2, splitLength)

	union := make(map[string]int)
	for i := range splittedStr1 {
		union[i] = 0
	}
	for i := range splittedStr2 {
		union[i] = 0
	}

	res := 0

	for i := range union {
		res += int(math.Abs(float64(splittedStr1[i] - splittedStr2[i])))
	}

	return res
}

func Shingle(s string, k int) map[string]int {
	m := make(map[string]int)
	if s != "" && k != 0 {
		runeS := []rune(s)

		for i := 0; i < len(runeS)-k+1; i++ {
			m[string(runeS[i:i+k])]++
		}
	}
	return m
}

File: translations_only/fail/go/go-edlib/QgramDistance/claude2/Hinted/QgramDistance.rs

use std::collections::HashMap;

fn qgram_distance(str1: &str, str2: &str, split_length: isize) -> isize {
    let splitted_str1 = shingle(str1, split_length);
    let splitted_str2 = shingle(str2, split_length);

    let mut union = HashMap::new();
    for i in splitted_str1.keys() {
        union.insert(i, 0);
    }
    for i in splitted_str2.keys() {
        union.insert(i, 0);
    }

    let mut res = 0;

    for count in union.values_mut() {
        res += abs_diff(count, 0) as isize;
    }

    res
}

fn abs_diff(a: &isize, b: isize) -> isize {
    let diff = (a - b).abs();
    diff
}

fn shingle(s: &str, k: isize) -> HashMap<String, isize> {
    let mut m = HashMap::new();
    if !s.is_empty() && k != 0 {
        let rune_s: Vec<_> = s.chars().collect();

        for i in 0..(rune_s.len() - k as usize + 1) {
            *m.entry(rune_s[i..(i + k as usize)].iter().collect()).or_insert(0) += 1;
        }
    }
    m
}

